pkg/api/configsync/v1beta1/resource_override.go

```diff
@@ -66,6 +66,10 @@ type OverrideSpec struct {
 	// support pulling remote bases from public repositories.
 	// +optional
 	EnableShellInRendering *bool `json:"enableShellInRendering,omitempty"`
+
+	// logLevel allows one to override the log verbosity of a container
+	// +optional
+	LogLevel map[string]int `json:"logLevel,omitempty"`
```

Comment:
I think this should cause the reconciler to stall with a validation error if the user provides an unexpected container name

Reply:
Alternatively, we could enumerate the valid containers as a nested CRD field. Might want to get agreement on the API in a design doc

Reply:
from my understanding, the iteration is done in https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/reconcilermanager/controllers/rootsync_controller.go#L917, and it will check for the container name in the override. Unexpected container name should be ignored. 

I tried reproducing this scenario in a cluster with local OSS installation with invalid container name, not seeing any error in yaml validation and reconciler log

Reply:
It is ignored currently, I'm suggesting that the behavior should be added to cause the reconciler to stall if the container name specified for this field is invalid

Reply:
updated to use a struct with containerName string and logLevel int field